app/src/main/java/com/example/sony/training/MainActivity.java

```diff
@@ -1,13 +1,296 @@
+import android.app.Activity;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.content.SharedPreferences.Editor;
+
+public class MainActivity extends Activity
+        implements OnEditorActionListener, OnSeekBarChangeListener,
+        OnCheckedChangeListener, OnItemSelectedListener, OnKeyListener {
+
+    // define variables for the widgets
+    private EditText billAmountEditText;
+    private TextView percentTextView;
+    private SeekBar percentSeekBar;
+    private TextView tipTextView;
+    private TextView totalTextView;
+    private RadioGroup roundingRadioGroup;
+    private RadioButton roundNoneRadioButton;
+    private RadioButton roundTipRadioButton;
+    private RadioButton roundTotalRadioButton;
+    private Spinner splitSpinner;
+    private TextView perPersonLabel;
+    private TextView perPersonTextView;
+
+    // define the SharedPreferences object
+    private SharedPreferences savedValues;
 
-public class MainActivity extends AppCompatActivity {
+    // define rounding constants
+    private final int ROUND_NONE = 0;
+    private final int ROUND_TIP = 1;
+    private final int ROUND_TOTAL = 2;
+
+    // define instance variables
+    private String billAmountString = "";
+    private float tipPercent = .15f;
+    private int rounding = ROUND_NONE;
+    private int split = 1;
 
     @Override
-    protected void onCreate(Bundle savedInstanceState) {
+    public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        // get references to the widgets
+        billAmountEditText = (EditText) findViewById(R.id.billAmountEditText);
+        percentTextView = (TextView) findViewById(R.id.percentTextView);
+        percentSeekBar = (SeekBar) findViewById(R.id.percentSeekBar);
+        tipTextView = (TextView) findViewById(R.id.tipTextView);
+        totalTextView = (TextView) findViewById(R.id.totalTextView);
+        roundingRadioGroup = (RadioGroup)
+                findViewById(R.id.roundingRadioGroup);
+        roundNoneRadioButton = (RadioButton)
+                findViewById(R.id.roundNoneRadioButton);
+        roundTipRadioButton = (RadioButton)
+                findViewById(R.id.roundTipRadioButton);
+        roundTotalRadioButton = (RadioButton)
+                findViewById(R.id.roundTotalRadioButton);
+        splitSpinner = (Spinner) findViewById(R.id.splitSpinner);
+        perPersonLabel = (TextView) findViewById(R.id.perPersonLabel);
+        perPersonTextView = (TextView) findViewById(R.id.perPersonTextView);
+
+        // set array adapter for spinner
+        ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFromResource(
+                this, R.array.split_array, android.R.layout.simple_spinner_item);
+        adapter.setDropDownViewResource(
+                android.R.layout.simple_spinner_dropdown_item);
+        splitSpinner.setAdapter(adapter);
```

Comment:
Pls define "iniViews" method and move this code (line 66 -> 88) into this method